src/main/java/core/basesyntax/service/FruitShopDataProcessorService.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax.service;
+
+import core.basesyntax.handler.ShopOperationHandler;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.strategy.ShopOperationStrategy;
+import core.basesyntax.utility.FruitType;
+import java.util.List;
+import java.util.Map;
+
+public class FruitShopDataProcessorService implements DataProcessorService {
+    public static final int OPERATION_INDEX = 0;
+    public static final int FRUIT_INDEX = 1;
+    public static final int QUANTITY_INDEX = 2;
+    private final ShopOperationStrategy shopOperationStrategy;
+    private final Map<FruitType, Fruit> fruitMap;
+
+    public FruitShopDataProcessorService(ShopOperationStrategy shopOperationStrategy,
+                                         Map<FruitType, Fruit> fruitMap) {
```

Comment:
It's not good practice to transfer services/utils via method arguments. By fact it doesn't required here, it just taken to be passed to handler.
```suggestion
    public FruitShopDataProcessorService(ShopOperationStrategy shopOperationStrategy) {
```

Please use your Storage as constructor argument for handlers or provide static method to it so handlers may access it without depenency